config/config.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+return [
+    'bookings_front_ui' => [
+        'holder_template' => '<div data-eddbk-wizard-holder><eddbk-wizard :config=\'%s\'></eddbk-wizard></div>',
+        'edd_settings' => get_option('edd_settings'),
+
+        /*
+         * Configuration for placeholder templates.
+         */
+        'templates_config' => [
+            'token_start'   => '${',
+            'token_end'     => '}',
+            'token_default' => '',
+        ],
+
+        'application_selector' => '[data-eddbk-wizard-holder]',
+        'api_base_urls' => [
+            'bookings' => '/eddbk/v1/bookings/',
+            'services' => '/eddbk/v1/services/',
+        ],
+        'initial_booking_transition' => 'cart',
+        'formats' => [
+            'datetime' => [
+                'sessionTime' => 'h:mm a',
+                'dayFull' => 'dddd Do MMMM YYYY',
+                'dayShort' => 'D MMMM YYYY',
+                'dayKey' => 'YYYY-MM-DD',
+                'appointmentStart' => 'h:mm a \o\\n dddd Do MMMM YYYY'
+            ]
+        ],
+
+        'assets_urls_map_path' => WP_BOOKINGS_FRONT_UI_MODULE_CONFIG_DIR . '/assets_urls_map.php',

```

Comment:
I've been thinking about how to avoid these constants in configs for a long time. We know that we can use config path placeholders, which will be replaced with the config value upon retrieval. Maybe, we should hook in a special implementation, which would map constant names to their values? This will allow us to avoid using PHP here, as well as to override those values (as opposed to constants, which cannot be re-defined). I would also like to hear from @mecha on this subject.

Reply:
Or maybe we should try to avoid using constants altogether?

Reply:
I'd rather not use constants than add a mechanism for simply mapping the constant name to its value.

But I'd also like to point out that this is _still_ PHP config, and it's also _module_ config. While I acknowledge that using constant names here is not ideal, it's not wrong either. If we were using XML or JSON, we'd need alternatives for sure. But this works and (IMO) can stay for now. Just expressing my 2 cents - I'm really not looking to debate this.

If I absolutely _had_ to remove constants from config, I'd probably just replace the constants in `module.php` with an array map, inject it into the module, and that module would use that as the base of the config, effectively merging it with the config from file. This would allow token replacement.

Reply:
Discussed this in person. We should do this, but later.